File: snail-race-bet/typescript/src/domain/Bet.ts
export class Bet {
    constructor(
        public gambler : string,
        public pronostic :  PodiumPronostic,
        public timestamp : number
    ) {
    }
}

export class PodiumPronostic {
    constructor(
        public readonly first: number,
        public readonly second: number,
        public readonly third: number
    ) {

    }
}

File: snail-race-bet/java/src/main/java/snail/race/kata/domain/Bet.java
package snail.race.kata.domain;

public record Bet(
        String gambler,
        PodiumPronostic pronostic,
        long timestamp
){

    public boolean isInTimeFor(RaceResultProvider.SnailRace race) {
        return timestamp > race.timestamp() - 2;
    }

    boolean betIsOn(RaceResultProvider.Podium podium) {
        return pronostic().first() == podium.first().number() &&
               pronostic().second() == podium.second().number() &&
               pronostic().third() == podium.third().number();
    }
}

